src/components/atoms/IconInput/IconInput.scss

```diff
@@ -0,0 +1,41 @@
+@import "scss/constants";
+
+.icon-input {
+  position: relative;
+
+  &__input {
+    height: 100%;
+
+    &--icon-from-left {
+      padding-left: 2.5rem;
+    }
+
+    &--icon-from-right {
+      padding-right: 2.5rem;
+    }
+
+    &:focus + .icon-input__icon {
+      opacity: 0.6;
+    }
+  }
+
+  &__icon {
+    display: flex;
+    position: absolute;
+    top: 0;
+    height: 100%;
+    width: 40px;
+    justify-content: center;
+    align-items: center;
+    z-index: z(input-icon);
+    opacity: 0.2;
+
+    &--left {
+      left: 0;
+    }
+
+    &--right {
+      right: 0;
+    }
```

Comment:
We could potentially move these variants onto the base class itself, so that they are:

- `.icon-input--icon-left`
- `.icon-input--icon-right`

or similar. Not sure if that is better or worse than how it is now though (@mike-lvov any thoughts?)

Then these would end up being written as something like (in the 'root' of the main class):

```scss
&--icon-left {
  .icon-input__icon {
    left: 0;
  }
}

&--icon-right {
  .icon-input__icon {
    right: 0;
  }
}
```

This would also allow any of the other style changes related to the variant to be 'grouped together' in this 'single location', and then we would only need a single check in the `classNames` helper to set the variant on the root component's classes.

Reply:
We potentially could even drop the word 'icon' from this, and just have the variants as `--left` and `--right`?

Reply:
added `--icon-star` and `--icon-end` modificators to the base `input-field` class. 

commit: https://github.com/sparkletown/sparkle/pull/1306/commits/4522f27ae790a9bc30e71c2e4e06433f7e2b9141